### ISSUE TITLE
Ensure uniform access for all config options

### ISSUE THREAD

**Issue by jdavcs** [galaxyproject/galaxy#14369]
This is a follow-up on #14364.

The `GalaxyAppConfiguration` object provides methods `get` and `get_bool` which are used to retrieve the current value of a configuration property, or provide a default if the option is not set.

```py
foo = self.app.config.get('config-property-foo', some-default-value-for-foo)
```

Both methods obtain the values they return from `config_dict` - an instance variable on the same object that is a copy of the keyword argument dictionary it [gets on construction](https://github.com/galaxyproject/galaxy/blob/release_22.05/lib/galaxy/config/__init__.py#L214). HOWEVER, this may be the incorrect value, as the loading of config values has undergone significant changes in 2019 (e.g. default values are loaded from `config_schema.yaml`, some relative paths are automatically resolved w.r.t. their parent paths, some values may be reloaded at runtime, etc. - these are the main changes, but there were more; ref #8475, #8689, #8639, #8573; also see issue #8493).

There are many instances of such usage in the code base. In some cases it facilitates retrieving a non-documented property that would have been set as an attribute on the config object only if it were passed at startup. In others, it may facilitate a testing scenario where the config_dict is constructed in the testing setup code. Both of these use cases can be eliminated with (a) adding non-documented options to the schema (in my opinion, if they are referenced in code, they must be documented); and (b) some minor refactoring of the testing setup code. Another (even simpler) alternative is to ensure both methods use the appropriate config object's attribute, not `config_dict` (initial kwgs), and iff that attribute is not defined, then return the provided default - but that would be justifiable only if the previous 2 options are untenable.

**Comment by mvdbeek** [comment]
Have you considered building an actual typed config model ? It would certainly be super-helpful in catching issues there, and I think we can generate that from the existing schema.

**Comment by jdavcs** [comment]
> Have you considered building an actual typed config model ? It would certainly be super-helpful in catching issues there, and I think we can generate that from the existing schema.

Yes, I have. Maybe now is a good time to try it. I need to revisit my notes on this; I'll open a separate issue soon.
